### PR TITLE
IDCOM-1335 Updated W3C docs to include the generated DID document

### PR DESCRIPTION
**Pull request by explicit-projects**
Updated the W3C docs to explain how the generated DID documents are used in the following cases:
* Create
* Read
* Delete

(Some minor formatting udates)

### CODE REVIEW

File: docs/did-method-spec.html [hunk 1]
```diff
@@ -119,51 +119,54 @@ <h3>Examples</h3>
...
+        }
+    ]
+}
 </pre>
   where: </p><p>
   <code>AUTHORITY</code> is the base58-encoded 32-byte x25519 public key of the DID owner,</p><p>
   <code>IDENTIFIER</code> is the DID method identifier derived from the authority.</p><p>
 
-  The transaction must be signed by the authority.
+    Registering a SOL DID involves submitting a transaction to the Solana blockchain calling the "Initialize" instruction
```
Comment (dankelleher) [inline docs/did-method-spec.html:165]: Change registering to updating here? Or "changing the default data"

File: docs/did-method-spec.html [hunk 2]
```diff
@@ -194,7 +197,7 @@ <h4 id="fees">Fees and Rent</h4>
     </li>
   </ul>
   <p>By default, the fee payer is the owner of the DID that will be created. However, the implementation supports
-  a separate fee payer, that will incur the charges of creation/updating/deletion of the DID without having
+  a separate fee payer, that will incur the charges of registration/updating/deletion of the DID without having
   any permissions on the DID itself. This allows a central body to generate DIDs for its customers, without
   requiring them to have a SOL balance.</p>
   <h3>Read (Resolve)</h3>
```
Comment (dankelleher) [inline docs/did-method-spec.html:203]: "<p>A SOL DID record can be looked up on Solana,by stripping the did:sol: prefix from the DID, and
    network identifier if present, and looking up the account on-chain.</p>"

I would rephrase this to:

"<p>A SOL DID record can be looked up on Solana,by stripping the did:sol: prefix from the DID, and
    network identifier if present, and looking up the resultant account on-chain. If no data is present on-chain, the generated DID document (see above) should be returned.</p>

File: docs/did-method-spec.html [hunk 3]
```diff
@@ -208,6 +211,7 @@ <h3>Read (Resolve)</h3>
     The schema for this format is defined in the
     <a href="#reference-implementations">reference implementation client.</a>
   </p>
+    <p>If the DID is not present on on-chain, the <a href="create">generated</a> DID document must be used.</p>
```
Comment (dankelleher) [inline docs/did-method-spec.html:214]: can be merged into the above.

File: docs/did-method-spec.html (hunk 2 above)
Comment (dankelleher) [reply docs/did-method-spec.html:203]: I would also suggest adding details about the default information. Specifically something like:

The authority is a special key from which the DID is derived. It is always present in the verification methods list, and by default is added as the only key in the capabilityInvocations list, indicating that it is permitted to change the DID Document contents. It can, however be removed from the list by submitting an update instruction where the capabilityInvocation list is explicitly set (e.g. to the empty list or some other list of keys)

Also add links to the above where appropriate e.g. around capabilityInvocation pointing to the w3c did document spec and around update instruction pointing to the update section.

**Comment by explicit-projects** [comment]
@dankelleher I feel like I got tripped up by the headings initially (espescially since trying to fit the generative case into the existing docs). 

We have:
* Create (Register)
* Read (Resolve)
* Update (Replace)
* Delete (Revoke)

In the W3C spec, they have (still CRUD):
* Create
* Resolve
* Update
* Deactivate

It feels more natural to include the generative case on the "create" (and not mix it with 'registering')... explain the on-chain part in the update and delete, and seperate the "Fees and Rent" into it's own section above "Security Considerations". 

Mostly just structure (copy/paste) change - and incorporating your above feedback

Lemme know ?

**Review by dankelleher — APPROVED** [review]
